brave/src/main/java/com/linecorp/armeria/client/brave/ClientRequestContextAdapter.java

```diff
@@ -74,7 +74,11 @@ public String path() {
         @Override
         @Nullable
         public String url() {
-            return SpanTags.generateUrl(ctx.log());
+            // This dies because lack of RequestLogAvailability.REQUEST_HEADERS
+            //return SpanTags.generateUrl(ctx.log());
+
+            // This doesn't die
+            return ((HttpRequest) ctx.request()).uri().toString();
```

Comment:
PS I don't mind doing this instead, just the casting seems accidental/awkward

Reply:
IIRC `ClientRequestContext#request` returns `RpcRequest` when it's RpcClient.

Reply:
IIRC `ctx.request` returns `RpcRequest` when it's a rpc client.

Reply:
ok so maybe we should add a test which ends up returning an RpcRequest to ensure we don't break it. Do you feel up to adding a commit for that?

Reply:
ok,  I will make one in another PR.

Reply:
I'm thinking of splitting `request()` into `request(): HttpRequest` and `rpcRequest(): RpcRequest`, which is similar to @anuraaga's suggestion. Please stay tuned.